### ISSUE TITLE
Manual graph replay produces garbled output (590 kernels recorded)

### ISSUE THREAD

**Issue by noahgift** [paiml/realizar#198]
## Context

PMAT-453: Manual CUDA graph construction via `cuGraphAddKernelNode` (trueno#243)
bypasses stream capture (blocked by code 901 on driver 570.207).

## What works

- **590 kernel nodes recorded** during eager decode forward pass (was 0)
- Graph builds successfully via `cuGraphCreate` + `cuGraphAddKernelNode`
- `cuGraphInstantiateWithFlags` succeeds
- All decode-path kernels wired: RMSNorm, DP4A GEMV, RoPE indirect, attention,
  KV scatter indirect, flash decoding, fused SwiGLU, Q8 quantize, residual add

## What's broken

Graph replay via `cuGraphLaunch` produces garbled output:
- **Baseline** (no graph): "I'm sorry, but I'm not sure what you're asking for..."
- **Graph replay**: "I'm'm'm'm'm'm'm'm'm'm'm'm..."

First token (eager path) is correct. Graph replay tokens degenerate into repetition.

## Likely root causes

1. **Arg value encoding**: u32 dimension args stored as u64 in `RecordedKernel::arg_data`.
   On little-endian x86, CUDA reads correct 4 bytes, but needs verification.
2. **Q8 cache conditional**: During recording, Q8 quantize is SKIPPED for K/V/Up projections
   (cache valid). Graph replays the recorded subset. Should be correct (shared input) but
   needs verification.
3. **cuGraphExecUpdate needed**: Per-token arg changes (different layer K/V cache pointers
   across layers) may require `cuGraphExecUpdate` instead of replaying the same graph.

## Reproduction

```bash
CUDA_GRAPH_ENABLE=1 realizar serve \
  --model qwen2.5-coder-1.5b-instruct-q4_k_m.gguf \
  --gpu --openai-api --port 8199
# Send any chat completion request with max_tokens > 1
```

## Commits

- 82512d66: Wire all decode-path kernels into manual graph recording
- 6ae0703d: Manual graph construction infrastructure

**Comment by noahgift** [comment]
## A/B Test Results: RMSNorm correct, divergence in layer pipeline

Replayed graph at SAME position as eager capture. Results:

| Buffer | Max Diff | Verdict |
|--------|----------|---------|
| hidden_buf1 (RMSNorm L0) | **0.000** | IDENTICAL |
| hidden_buf2 (28 layers) | **155.1** | DIVERGED |
| normed_hidden | **50.0** | DIVERGED |
| logits | **17.5** | DIVERGED |

**Conclusion:** Graph's first kernel (RMSNorm) produces IDENTICAL output.
Divergence happens inside the 28-layer pipeline (Q8→GEMV→RoPE→attention→FFN).

619 kernels = 28 × 22/layer + 3. All recorded, all launch configs correct.
Need binary search through layers (disable layers 14-27, check if first 14 match).

Commits: 06357373 (A/B diagnostic)

**Comment by noahgift** [comment]
## Root cause found and fixed (2797c878)

**Five-whys:**
1. Why garbled output? → hidden_buf2 diverges (81.0 max diff) between eager and graph replay
2. Why hidden_buf2 diverges? → input_staging carries wrong residual from attention, even though FFN output is correct
3. Why is FFN correct but residual wrong? → FFN buffers retain eager values (never overwritten by graph)
4. Why aren't FFN buffers overwritten? → The fused gate+up+SwiGLU kernel is NOT in the graph
5. Why isn't it recorded? → `fused_gate_up_swiglu_hw_dp4a_q4k_gemv_into` was **missing the `if self.graph_recording` block** — the only kernel function in the decode hot path without recording

**Fix:** Added graph recording to:
- `fused_gate_up_swiglu_hw_dp4a_q4k_gemv_into` (28 kernels/forward, the root cause)
- `mwv_dp4a_q4k_gemv_into` (backup GEMV variant, also missing)

**Results:**
- Graph: 647 kernels (was 619, +28 fused SwiGLU)
- A/B test: ALL 13 workspace buffers diff=0.000000
- Logits argmax: match=true
- Output: coherent ("together. The function should take two parameters")

Also enhanced the A/B diagnostic with per-buffer divergence scan for future debugging.

**Comment by noahgift** [comment]
Fixed in 0e4e9db7. Five-whys root cause: missing graph recording in fused_gate_up_swiglu_hw_dp4a_q4k_gemv_into. 647 kernels now recorded (was 619). A/B test: ALL 13 buffers diff=0. **331.2 tok/s decode** (+26.3% vs eager).
